src/datatypes/dtype.rs

```diff
@@ -75,8 +75,12 @@ pub enum DataType {
     /// Extension type.
     Extension(String, Box<DataType>, Option<String>),
     // Stop ArrowTypes
-    /// A Embedding Logical Type
+    /// A logical type for embeddings.
     Embedding(Box<Field>, usize),
+    /// A logical type for images with variable shapes.
+    Image(Box<Field>),
+    /// A logical type for images with the same shape.
+    FixedShapeImage(Box<Field>, Vec<usize>),
```

Comment:
I'm thinking that keeping these as two separate types internally would be good, since we'll be able to support a lot more ops for the `FixedShapeImage` type than the ragged `Image` type, and even for where there's overlap, we'll want deviating implementations to take advantage of being able to vectorize on the row dimension for the former.

Reply:
okay that sounds good!

Reply:
left some comments on the review to use dim names instead of a generic shape